cmd/hypersdk-cli/transaction.go

```diff
@@ -0,0 +1,224 @@
+			if txErr != nil {
+				return txErr
+			}
+			if txID == expectedTxID {
+				result = txResult
+				break
+			}
+		}
+
+		var resultStruct map[string]interface{}
+		if result.Success {
+			if len(result.Outputs) == 1 {
+				resultJSON, err := dynamic.UnmarshalOutput(abi, result.Outputs[0])
+				if err != nil {
+					return fmt.Errorf("failed to unmarshal result: %w", err)
+				}
+
+				err = json.Unmarshal([]byte(resultJSON), &resultStruct)
+				if err != nil {
+					return fmt.Errorf("failed to unmarshal result JSON: %w", err)
+				}
+			} else if len(result.Outputs) > 1 {
+				return fmt.Errorf("expected 1 output, got %d", len(result.Outputs))
+			}
+		}
+
+		return printValue(cmd, txResponse{
+			Result:  resultStruct,
+			Success: result.Success,
+			TxID:    expectedTxID,
+			Error:   string(result.Error),
+		})
+	},
+}
+
+type txResponse struct {
+	Result  map[string]interface{} `json:"result"`
+	Success bool                   `json:"success"`
+	TxID    ids.ID                 `json:"txId"`
+	Error   string                 `json:"error"`
+}
+
+func (r txResponse) String() string {
+	var result strings.Builder
+	if r.Success {
+		result.WriteString(fmt.Sprintf("✅ Transaction successful (txID: %s)\n\n", r.TxID))
+		if r.Result != nil {
+			for key, value := range r.Result {
+				jsonValue, err := json.Marshal(value)
+				if err != nil {
+					jsonValue = []byte(fmt.Sprintf("%v", value))
+				}
+				result.WriteString(fmt.Sprintf("%s: %s\n", key, string(jsonValue)))
+			}
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("❌ Transaction failed (txID: %s): %s\n", r.TxID, r.Error))
+	}
+	return strings.TrimSpace(result.String())
+}
+
+func init() {
+	txCmd.Flags().StringToString("data", nil, "Key-value pairs for the action data (e.g., key1=value1,key2=value2)")
+	rootCmd.AddCommand(txCmd)
+}
+
+func SignTxManually(actionsTxBytes [][]byte, base *chain.Base, privateKey ed25519.PrivateKey) ([]byte, error) {
+	// Create auth factory
+	factory := auth.NewED25519Factory(privateKey)
+
+	// Marshal base
+	p := codec.NewWriter(base.Size(), consts.NetworkSizeLimit)
+	base.Marshal(p)
+	baseBytes := p.Bytes()
+
+	// Build unsigned bytes starting with base and number of actions
+	unsignedBytes := make([]byte, 0)
+	unsignedBytes = append(unsignedBytes, baseBytes...)
+	unsignedBytes = append(unsignedBytes, byte(len(actionsTxBytes))) // Number of actions
+
+	// Append each action's bytes
+	for _, actionBytes := range actionsTxBytes {
+		unsignedBytes = append(unsignedBytes, actionBytes...)
+	}
+
+	// Sign the transaction
+	auth, err := factory.Sign(unsignedBytes)
+	if err != nil {
+		return nil, err
+	}
+	// Marshal auth
+	p = codec.NewWriter(auth.Size(), consts.NetworkSizeLimit)
+	auth.Marshal(p)
+	authBytes := []byte{auth.GetTypeID()}
+	authBytes = append(authBytes, p.Bytes()...)
+
+	// Combine everything into final signed transaction
+	//nolint:gocritic //append is fine here
+	signedBytes := append(unsignedBytes, authBytes...)
+	return signedBytes, nil
+}
```

Comment:
Rather than having tx bytes construction logic here, could we add the required helper into `chain/transaction.go` ?